src/geh_common/telemetry/decorators.py

```diff
@@ -21,3 +30,50 @@ def wrapper(*args: Tuple[Any], **kwargs: Dict[str, Any]) -> Any:
         return wrapper
 
     return decorator
+
+
+def start_trace(initial_span_name: str | None = None) -> Callable[..., Any]:
+    """Set up initial span.
+
+    Decorator that checks if the logging_configuration.configure_logging method has been called prior to starting the
+    trace. Provides an initial span based on the provided initial_span_name parameter.
+    """
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        func_signature = inspect.signature(func)  # Get function signature of function applying the decorator
+        accepts_initial_span = (
+            "initial_span" in func_signature.parameters
+        )  # Check if initial_span is in function params
+
+        def wrapper(*args: Tuple[Any], **kwargs: Dict[str, Any]) -> Any:
+            # Retrieve the logging_configured flag from logging_configuration to see if configure_logging() has been called
+            logging_configured = get_logging_configured()
+            name_to_use = initial_span_name or func.__name__
+            if logging_configured:
```

Comment:
Reduce nesting by inverting the if condition. Read more [here](https://en.wikipedia.org/wiki/Guard_(computer_science)#Flatter_code_with_less_nesting).